lib/ace/edit_session.js

```diff
@@ -324,6 +324,19 @@ var EditSession = function(text, mode) {
     };
 
     /**
+     * @param {Number} row
+     * 
+     * Returns context at row.
+     **/
+    this.getContext = function(row) {
```

Comment:
There is `getState` method above. I think we can move the analysis of the returned state to our own code